Fix: Ensure consistent underlining of chatBTC links (#1214)

rajanarahul93 opened the pull request:
This PR resolves issue #1214  by ensuring consistent formatting for chatBTC links. Previously, some links were underlined while others were not. This update:

✅ Applies a consistent underline style to all chatBTC links
✅ Refactors code to prevent future inconsistencies
✅ Uses reusable components for better maintainability

Changes Made:
Updated HolocatQuestion.tsx to apply a uniform text style.
Ensured all links either have an underline or none (as per team preference).
Refactored tooltip and inline styling to avoid manual overrides.
Testing Done:
✅ Manually tested on different pages.
✅ Verified tooltip and inline styles render correctly.

0tuedon commented on package-lock.json:
Hi @rajanarahul93 , Thanks for raise a PR for this issue,  currently we use yarn and we do not need to have a new package-lock.json.

rajanarahul93 replied:
Okay

0tuedon commented on yarn.lock:
You could exclude the changes here too, we don't need the yarn.lock being modified if we didn't add, update or remove a new package.

rajanarahul93 replied:
Yeah,sure!!

rajanarahul93 commented:
Hi @0tuedon ,
I’ve removed package-lock.json and reverted changes to yarn.lock as requested. Let me know if anything else needs to be updated. Thanks!

0tuedon commented on ui/common/HolocatQuestion.tsx:
Thank you again for making this PR, I do really appreciate the effort, I have been able to review each page where this component is used, and currently here are the pages which are not still underline

- https://saving-satoshi-3lnrfxb45-savingsatoshi.vercel.app/en/chapters/chapter-4/address-1?dev=true

- https://saving-satoshi-3lnrfxb45-savingsatoshi.vercel.app/en/chapters/chapter-6/in-out-4-normal?dev=true (little-endian and reversed) holocat questions

- https://saving-satoshi-3lnrfxb45-savingsatoshi.vercel.app/en/chapters/chapter-7/mempool-transaction-1?dev=true (transaction weights and order of transactions)

- https://saving-satoshi-3lnrfxb45-savingsatoshi.vercel.app/en/chapters/chapter-8/building-blocks-1?dev=true (light-clients)

rajanarahul93 replied:
Sorry for the inconvenience!! I will double check the files and push it again

rajanarahul93 commented:
I have applied consistent underlining to all chatBTC links, including the inline tooltip case. Please review and let me know if further adjustments are needed. Thanks!
@0tuedon

0tuedon reviewed (changes requested):
Great work! @rajanarahul93, one final thing could you squash the commits into one or two, and I can see some duplicate commit message. LMK if you need help doing that.

rajanarahul93 commented:
Yeah sure, i will try to do it

rajanarahul93 commented:
I tried squashing all the commits but faced some difficulties. Could you guide me on how to do it?

rajanarahul93 commented:
Hi @0tuedon ,

I have successfully squashed the commits into a single commit and pushed the changes. Let me know if any further adjustments are needed.

0tuedon reviewed (changes requested):
Great work @rajanarahul93 , I think we have our yarn.lock problem back.

0tuedon commented:
> I tried squashing all the commits but faced some difficulties. Could you guide me on how to do it? @0tuedon

What issues are you having?

rajanarahul93 commented:
I got it know , i pushed the squashed commit > > I tried squashing all the commits but faced some difficulties. Could you guide me on how to do it? @0tuedon
> 
> What issues are you having?

rajanarahul93 commented:
Hi @0tuedon ,

I have successfully squashed the commits into a two commit and also reverted the unintended changes to yarn.lock. Please review the latest update and let me know if any further modifications are needed.

rajanarahul93 commented:
Hi @0tuedon ,

I’ve successfully squashed the commits into two and reverted the unintended changes to yarn.lock. Could you check if all issues are now resolved? If everything looks good, can you proceed with merging the PR?

0tuedon reviewed (approved):
Great work @rajanarahul93 , LGTM!. Thank you so  much for your Contribution.

0tuedon reviewed (commented):
Could you squash these commits again, i am having issues merging this. @rajanarahul93 .

rajanarahul93 commented:
![image](https://github.com/user-attachments/assets/db98fc54-2c8e-406b-9587-f8cdd6ecc18f)
should i squash these two into one ?

0tuedon commented:
>  should i squash these two into one ?

No, Just rebase and force push again.

<img width="877" alt="Screenshot 2025-03-21 at 19 17 16" src="https://github.com/user-attachments/assets/da29b57c-8781-4ac3-92f2-99468cf340d5" />

rajanarahul93 commented:
Hi @0tuedon ,
I have done the rebase and force push once again. Please check if the issue is resolved now. Let me know if any further action is needed. Thanks!

0tuedon commented:
> Hi @0tuedon , I have done the rebase and force push once again. Please check if the issue is resolved now. Let me know if any further action is needed. Thanks!

Sure could you try adding an empty commit.

rajanarahul93 commented:
Hi @0tuedon ,
I have added an empty commit as you suggested and pushed the changes again. Could you check if the issue is resolved now?

0tuedon commented:
> Hi @0tuedon , I have added an empty commit as you suggested and pushed the changes again. Could you check if the issue is resolved now?

Good, it has been resolved.

rajanarahul93 commented:
> > Hi @0tuedon , I have added an empty commit as you suggested and pushed the changes again. Could you check if the issue is resolved now?
> 
> Good, it has been resolved.

Great!!
